native/src/main/java/io/ballerina/stdlib/websocket/actions/websocketconnector/WebSocketSyncConnector.java

```diff
@@ -52,6 +56,9 @@ private static void readContentFromConnection(BObject wsConnection, Future callb
                 WebSocketConstants.ANNOTATION_ATTR_READ_IDLE_TIMEOUT, 0);
         connectionInfo.getWebSocketConnection().addReadIdleStateHandler(readTimeoutInSeconds);
         connectorListener.setCallback(callback);
+        if (targetType.length > 0) {
+            connectorListener.setTargetType(targetType[0]);
```

Comment:
What happens to the other types sent along as typedescs? eg, targetType[1], targetType[2]

Reply:
`targetType` is taken as an optional parameter. We are always calling this function with one optional parameter or without the optional parameter. So there won't be `targetType[1], targetType[2]` values.